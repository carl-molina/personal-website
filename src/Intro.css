a {
  color: white;
  transition: background-color 0.33s ease;
}

a:hover {
  background-color: rgb(61, 129, 112);
}

a:visited {
  color: white;
}