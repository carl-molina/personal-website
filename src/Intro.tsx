import React from "react";
import "./Intro.css";

/** Intro: presentational component for bio.
 *
 * Props: none
 * State: none
 *
 * App -> Intro
 */

function Intro() {

    return (
        <>
            <h1>Software Engineer</h1>
            <h3>San Jose, CA</h3>
            <p>Former Field Op Specialist with calibration and sensor health
                experience in the autonomous vehicles industry.<br></br>
                <br></br>
                Now full-stack developer building React apps, automating tests,
                and&nbsp;<i>
                <a
                    href="https://medium.com/@carlandrew.molina
                    /ordering-a-burritozilla-in-the-middle-of-a-coding-bootcamp
                    -15dadf1eca22"
                    target="_blank"
                    rel="noopener"
                >
                    deciding whether or not to eat a Burritozilla for the
                    umpteenth time.
                </a></i>
            </p>
        </>
    )
}

export default Intro;
