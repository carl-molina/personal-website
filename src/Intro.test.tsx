import React from "react";
import { test, expect, describe } from "vitest";
import { render } from '@testing-library/react';

import Intro from './Intro';


describe("Intro smoke test", function () {
  test('renders without crashing', function () {
      render(<Intro />);
  });
});

describe('Intro renders correct data', function () {
  test('contains expected text', function () {
    const res = render(<Intro />);
    expect(res.queryByText('Software Engineer')).toBeInTheDocument();
    expect(res.queryByText('Sunnyvale, CA')).toBeInTheDocument();
  });
  test('does not contain other component text', function () {
    const res = render(<Intro />);
    expect(res.queryByText('carl@carlmolina.com')).not.toBeInTheDocument();
    expect(res.queryByText('Jobly 👩🏽‍💼')).not.toBeInTheDocument();
    expect(res.queryByText('Pix.ly 📷')).not.toBeInTheDocument();
    expect(res.queryByText('Plant App 🌱')).not.toBeInTheDocument();
  });
});


describe("Intro snapshot test", function () {
  test('matches snapshot', function () {
      const { container } = render(<Intro />);
      expect(container).toMatchSnapshot();
  });
});
