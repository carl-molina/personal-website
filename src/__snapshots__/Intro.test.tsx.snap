// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Intro snapshot test > matches snapshot 1`] = `
<div>
  <h1>
    Software Engineer
  </h1>
  <h3>
    Sunnyvale, CA
  </h3>
  <p>
    Former Field Op Specialist with calibration and sensor health experience in the autonomous vehicle industry.
    <br />
    <br />
    Now full-stack software developer building React-TypeScript apps, automating tests, andÂ 
    <i>
      <a
        class="Medium-post"
        href="https://medium.com/@carlandrew.molina
                    /ordering-a-burritozilla-in-the-middle-of-a-coding-bootcamp
                    -15dadf1eca22"
        rel="noopener"
        target="_blank"
      >
        deciding whether or not to eat a Burritozilla for the umpteenth time.
      </a>
    </i>
  </p>
</div>
`;
